frontend/helpers/generateFakeSurplusAuction.ts

```diff
@@ -0,0 +1,58 @@
+import { SurplusAuction, SurplusAuctionBase, SurplusAuctionStates } from 'auctions-core/src/types';
+import BigNumber from 'bignumber.js';
+import faker from 'faker';
+import { random } from 'lodash';
+
+const SURPLUS_AUCTION_STATES: SurplusAuctionStates[] = [
+    'just-started',
+    'have-bids',
+    'ready-for-collection',
+    'collected',
+    'requires-restart',
+];
+
+const NETWORKS = ['mainnet', 'kovan', 'goerli'];
+
+const generateFakeSurplusAuctionBase = function (): SurplusAuctionBase {
+    return {
+        id: faker.datatype.number(),
+        network: faker.helpers.randomize(NETWORKS),
+    };
+};
+
+export const generateFakeSurplusAuction = function (
+    state?: SurplusAuctionStates
+): SurplusAuction | SurplusAuctionBase {
+    const auctionBaseData = generateFakeSurplusAuctionBase();
+    const generatedState: SurplusAuctionStates = state || faker.helpers.randomize(SURPLUS_AUCTION_STATES);
+
+    if (generatedState === 'just-started') {
+        return auctionBaseData;
+    }
+    const receiveAmountDAI = new BigNumber(parseFloat(faker.finance.amount()));
+    const receiverAddress = faker.finance.ethereumAddress();
+    const auctionEndDate = faker.date.soon();
+    const bidEndDate = generatedState === 'have-bids' ? faker.date.recent() : undefined;
+    const earliestEndDate = bidEndDate
+        ? auctionEndDate.getUTCMilliseconds() > bidEndDate.getUTCMilliseconds()
+            ? bidEndDate
+            : auctionEndDate
+        : auctionEndDate;
+    const bidAmountMKR = new BigNumber(parseFloat(faker.finance.amount()));
+
+    return {
+        ...auctionBaseData,
+        network: 'mainnet',
+        receiveAmountDAI,
+        receiverAddress,
+        auctionEndDate,
+        bidEndDate,
+        earliestEndDate,
+        state: generatedState,
+        bidAmountMKR,
```

Comment:
One issue I saw in the types was that `bidAmountMKR` cannot be undefined. I see an issue here, as if the Auction State is `just-started` (auction is active, but has no bids) this would need to be undefined. What do you think?